src/sagas/bonding-curve.js

```diff
@@ -0,0 +1,67 @@
+import { takeLatest, call, select } from 'redux-saga/effects'
+
+import * as bondingCurveActions from '../actions/bonding-curve'
+import * as walletSelectors from '../reducers/wallet'
+import { bondingCurve } from '../bootstrap/dapp-api'
+import { lessduxSaga } from '../utils/saga'
+
+/**
+ * Fetch reserve parameters of the bonding curve.
+ * @returns {object} { totalETH, totalPNK, spread } all keys map to big number objects.
+ */
+function* fetchBondingCurveTotals() {
+  return {
+    totalETH: yield call(bondingCurve.getTotalETH),
+    totalPNK: yield call(bondingCurve.getTotalTKN),
+    spread: yield call(bondingCurve.getSpread)
+  }
+}
+
+/**
+ * Buy PNK from the bonding curve.
+ * @param {string} amount The amount of ETH the user has input.
+ * @returns {object} Updated reserve parameters of the bonding curve.
+ */
+function* buyPNKFromBondingCurve({ payload: { amount } }) {
+  const addr = yield select(walletSelectors.getAccount)
+  yield call(bondingCurve.buy, addr, 0, amount, addr)
+  return yield fetchBondingCurveTotals()
```

Comment:
Use `call`.

Reply:
Since `fetchBondingCurveTotals` is itself a saga generator and uses `call`, it should be OK.

Reply:
You are still supposed to call sagas with `call` for testability. See redux saga docs.